### PR TITLE
Add a command to list the peers in the node's Peer DB

### PR DESCRIPTION
**Pull request by mikethetike**
## Description

Add a `get_peers` command for the base node which lists the peers in its db 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by mikethetike** [comment]
I was also getting a number of errors when Bob's node would connect to Alice first in the tests, so I changed it so that Alice always dials Bob and Bob doesn't dial anyone. I updated the 3 node tests set up as well

File: base_layer/core/tests/helpers/nodes.rs [hunk 1]
```diff
@@ -285,20 +297,25 @@ pub fn create_network_with_2_base_nodes_with_config(
...
         .start(runtime, data_path);
 
-    let _ = runtime.block_on(
-        alice_node
-            .comms
-            .connection_manager()
-            .dial_peer(bob_node.node_identity.node_id().clone()),
-    );
+    // Wait for peers to connect
+    let res = runtime
```
Comment (sdbondi) [inline base_layer/core/tests/helpers/nodes.rs]: Unused var?

File: base_layer/core/tests/helpers/nodes.rs [hunk 2]
```diff
@@ -360,7 +377,7 @@ pub fn create_network_with_3_base_nodes_with_config(
         .start(runtime, data_path);
     let (bob_node, consensus_manager) = BaseNodeBuilder::new(network)
         .with_node_identity(bob_node_identity.clone())
-        .with_peers(vec![alice_node_identity.clone(), carol_node_identity.clone()])
+        .with_peers(vec![carol_node_identity.clone()])
```
Comment (sdbondi) [inline base_layer/core/tests/helpers/nodes.rs]: I agree with this change, Bob will know about Alice once Alice dials him so this prevents Bob dialling Alice at the same time - same for Carol :+1:

File: comms/src/connection_manager/dialer.rs [hunk 3]
```diff
@@ -460,7 +460,7 @@ where
                 },
                 // Delayed dial was cancelled
                 _ = cancel_signal => {
-                    debug!(target: LOG_TARGET, "[Attempt {}] Connecting to peer '{}'...", current_state.num_attempts(), current_state.peer.node_id.short_str());
+                    debug!(target: LOG_TARGET, "[Attempt {}] Connecting to peer '{}'... cancelled", current_state.num_attempts(), current_state.peer.node_id.short_str());
```
Comment (sdbondi) [inline comms/src/connection_manager/dialer.rs]:
```suggestion
                    debug!(target: LOG_TARGET, "[Attempt {}] Connection attempt cancelled for peer '{}'", current_state.num_attempts(), current_state.peer.node_id.short_str());
```

File: base_layer/core/tests/helpers/nodes.rs (hunk 1 above)
Comment (sdbondi) [reply base_layer/core/tests/helpers/nodes.rs]:
```suggestion
    runtime
```

File: base_layer/core/tests/helpers/nodes.rs [hunk 4]
```diff
@@ -249,10 +249,22 @@ pub fn create_network_with_2_base_nodes(
...
         .with_node_identity(bob_node_identity)
-        .with_peers(vec![alice_node_identity])
+        // Alice will call Bob, otherwise Bob may connect
+        // first and result in a DialCancelled
+        // .with_peers(vec![alice_node_identity])
         .with_consensus_manager(consensus_manager)
         .start(runtime, data_path);
 
+    // Wait for peers to connect
+    let res = runtime
```
Comment (sdbondi) [inline base_layer/core/tests/helpers/nodes.rs]:
```suggestion
    runtime
```

**Pull request merged**